File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        String str = br.readLine();

        int[] cnt = new int[26];
        //A 는 65번

        for(int i=0; i<str.length(); i++){
            cnt[(int)str.charAt(i)-65]++;
        }

        int oddCnt =0;
        boolean isOdd = str.length()%2!=0;
        boolean isPossible = true;
        for(int i=0; i<26; i++){
            if(cnt[i]%2!=0) oddCnt ++;
            if((isOdd && oddCnt>2)|| (!isOdd&&oddCnt>0)){
                isPossible=false;
                break;
            }
        }

        if(!isPossible) {
            System.out.println("I'm Sorry Hansoo");
            return;
        }

        char[] answer = new char[str.length()];
        int idx =0;
        for(int i=0; i<26; i++){
            if(cnt[i]%2!=0) {
                answer[str.length()/2] = (char)(i+65);
                cnt[i]--;
            }
            while(cnt[i]>0){
                answer[idx] = (char)(i+65);
                answer[str.length()-idx-1] = (char)(i+65);
                idx++;
                cnt[i]-=2;
            }
        }

        for(int i=0; i<str.length(); i++){
            sb.append(answer[i]);
        }

        System.out.println(sb.toString());
    }
}

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.py
word = input()

cnt_lst = [0]*26
a = ord('A')
l = len(word)
for s in word:
    cnt_lst[ord(s)-a] += 1

odd_cnt = 0
is_possible = True
for i in range(25):
    if cnt_lst[i] % 2 != 0:
        odd_cnt += 1
    if (l %2 ==0 and odd_cnt >=1) or ( l %2 !=0 and odd_cnt >=2):
        is_possible = False
        break

result = [0]*l


if is_possible:
    idx = 0
    for i in range(26):
        while cnt_lst[i] >= 2:
            if cnt_lst[i] > 1:
                result[idx] = chr(i+a)
                result[l-1 - idx] = chr(i+a)
                idx += 1
                cnt_lst[i] -= 2

        if cnt_lst[i] == 1:
            result[l // 2] = chr(i + a)
            cnt_lst[i] -= 1

    print(''.join(result))
else:
    print("I'm Sorry Hansoo")